syntax = "proto3";

package cognivault;

// Service definition
service CogniVaultService {
  // Simple ping method to test connectivity
  rpc Ping (PingRequest) returns (PingResponse);
  
  // Example method for processing data
  rpc ProcessData (ProcessDataRequest) returns (ProcessDataResponse);
}

// Request/Response messages
message PingRequest {
  string message = 1;
}

message PingResponse {
  string response = 1;
  int64 timestamp = 2;
}

message ProcessDataRequest {
  string data = 1;
  map<string, string> metadata = 2;
}

message ProcessDataResponse {
  string result = 1;
  bool success = 2;
  string message = 3;
}

